src/hotspot/cpu/ppc/macroAssembler_ppc.cpp

```diff
@@ -63,6 +63,28 @@
 bool AbstractAssembler::pd_check_instruction_mark() { return false; }
 #endif
 
+// Branch-free implementation to convert !=0 to 1
+// Set register dst to 1 if dst is non-zero. Use setbcr instruction on Power10.
+void MacroAssembler::normalize_bool(Register dst, Register temp, bool is_64bit) {
+
+  if (VM_Version::has_brw()) {
+    if (is_64bit) {
+      cmpdi(CCR0, dst, 0);
+    } else {
+      cmpwi(CCR0, dst, 0);
+    }
+    setbcr(dst, CCR0, Assembler::equal);
+  } else {
+    neg(temp, dst);
+    orr(temp, dst, temp);
+    if (is_64bit) {
+      srdi(dst, temp, 63);
+    } else {
+      srwi(dst, temp, 31);
+    }
+  }
+}
+
```

Comment:
If you want to place it below `MacroAssembler::set_cmpu3`: It's in macroAssembler_ppc.inline.hpp.